### PR TITLE
CUDA 11.3 support

### LINKED ISSUE #1293: CUDA 11.3 support
CUDA 11.3. is released and should be added.

There are some issues with CUDA 11.3

```
/alpaka/event/EventGenericThreads.hpp:280:19: error: ‘__T30’ was not declared in this scope
                 auto vQueues(dev.getAllQueues())
```

This can be solved by changing:

```C++
auto vQueues(dev.getAllQueues());
```
to
```C++
auto vQueues = dev.getAllQueues();
```

### PR DESCRIPTION
**Pull request by psychocoderHPC**
fix #1293 

- fix compile issues shown up with CUDA 11.3
- CI add CUDA 11.3
- update documentation

- [x] rebase against #1294

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ on: [push, pull_request]
 # ALPAKA_ACC_ANY_BT_OMP5_ENABLE                 : {ON, OFF}
 #   [ON] OMP_NUM_THREADS                        : {1, 2, 3, 4}
 # ALPAKA_ACC_GPU_CUDA_ENABLE                    : {ON, OFF}
-#   [ON] ALPAKA_CUDA_VERSION                    : {9.0, 9.1, 9.2, 10.0, 10.1, 10.2, 11.0}
+#   [ON] ALPAKA_CUDA_VERSION                    : {9.2, 10.0, 10.1, 10.2, 11.0, 11.1, 11.2, 11.3}
```
Comment (BenjaminW3) [inline .github/workflows/ci.yml]: We still test 9.0 and 9.1. Why did you delete them from the list?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -512,6 +521,28 @@ jobs:
...
+        - name: linux_nvcc-11.3_gcc-9_release
+          os: ubuntu-latest
+          env: { CXX: g++,     CC: gcc,    ALPAKA_CI_GCC_VER: 9,        ALPAKA_CI_STDLIB: libstdc++, CMAKE_BUILD_TYPE: Release, ALPAKA_CI_BOOST_BRANCH: boost-1.69.0, ALPAKA_CI_CMAKE_VER: 3.19.7,                     ALPAKA_CI_DOCKER_BASE_IMAGE_NAME: "ubuntu:18.04", ALPAKA_ACC_GPU_CUDA_ENABLE: ON, ALPAKA_CUDA_VERSION: "11.3", ALPAKA_CUDA_COMPILER: nvcc,                                                ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE: OFF }
+        - name: linux_nvcc-11.3_gcc-10_debug
+          os: ubuntu-latest
+          env: { CXX: g++,     CC: gcc,    ALPAKA_CI_GCC_VER: 10,       ALPAKA_CI_STDLIB: libstdc++, CMAKE_BUILD_TYPE: Debug,   ALPAKA_CI_BOOST_BRANCH: boost-1.73.0, ALPAKA_CI_CMAKE_VER: 3.20.0,                     ALPAKA_CI_DOCKER_BASE_IMAGE_NAME: "ubuntu:18.04", ALPAKA_ACC_GPU_CUDA_ENABLE: ON, ALPAKA_CUDA_VERSION: "11.3", ALPAKA_CUDA_COMPILER: nvcc, ALPAKA_CUDA_ARCH: "86",                        ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE: OFF }
+
```
Comment (BenjaminW3) [inline .github/workflows/ci.yml]: Shouldn't this block be directly above the new `# nvcc + clang++` block below?

**Review by BenjaminW3 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (psychocoderHPC) [reply .github/workflows/ci.yml]: I thought we need 9.2+, I will check again

File: .github/workflows/ci.yml (hunk 2 above)
Comment (psychocoderHPC) [reply .github/workflows/ci.yml]: Thanks, maybe I messed up with the order, I will check it again.

**Comment by sbastrakov** [comment]
Removed 0.6.1 project as on the VC we decided it will go to 0.7.0 and not be backported.

**Comment by psychocoderHPC** [comment]
> Removed 0.6.1 project as on the VC we decided it will go to 0.7.0 and not be backported.

Note: we decided in the developer meeting to aim for alpaka 0.7.0 feature freeze begin of May (2021) and to release 0.7.0 (with CUDA 11.3 support) latest around two months later.

**Review by sbastrakov — APPROVED** [review]

**Pull request merged**